scripts/get-version

```diff
@@ -8,7 +8,7 @@ fi
 
 ## PLEASE NOTE: This script is used primarily for use with homebrew and the "make brew" build target
 ## the logic to determine the version of pulumi at build time is now controlled via
-## https://github.com/pulumi/pulumictl and the `pulumictl get version` command
+## scripts/pulumi-version.sh and the PULUMI_VERSION environment variable.
```

Comment:
There's something going on in https://github.com/pulumi/pulumi/commits/master/scripts/get-version

It seems that we attempted to delete this script in favor of pulumictl to do versioning, then partially reverted that, but I can't find the reason. Looks like pulumictl version detection does not satisfy all the reqs here? 

Can we shed some light on this? 

Are we going to remove pulumictl now and maintain this script instead?

Would it possibly be better to patch pulumictl for the missing functionality? Happy to help hacking on this.

Reply:
The .version file is the canonical "release" version we are targeting in a repository, gives us control to edit the file to `3.x.1` if we wish to make a patch release.

`pulumictl get version` is based around git tags, tangentially this requires a deep clone resulting in slower checkouts, but more importantly, we aren't using git tags. All release builds will be on an untagged branch:

```
pulumictl get version --version-prefix 3.5.0
3.5.0-alpha.1662663630+52dcaaf7
```

I could patch `pulumictl get version` to take a literal version, but the shell script is much easier to understand at much lower complexity if we should like to change it.

Reply:
Whether the team prefers to maintain bash or Go tooling - we can figure out later. This should not be blocking the PR.